docs/README.md

```diff
@@ -1,2 +1,294 @@
+* DAG genertion : Takes several hours to run Ganesha has a machine to do this and shares the latest DAG info from Ropsten using [google drive](https://drive.google.com/file/d/1FqLCO5oc1xDYNMuub7xAqnb6kfohdf-U/view?usp=sharing). The epoch logic is the block Number divided by 30,000 so current Ropsten EPOCH is block 12280236 / 30000 = 409 which is the DAG info shared above. **Moving forward we need to update DAG information for every new EPOCH**
+* CLI Relayer : Relays the blocks, this is initially written in javascript as a Proof of concept and may be implemented in other languages moving forward. **Once the Relayer has begun we need to continually relay each block**
+* Client: The client is used to Process transactions this is done by locking the Token using the TokenLocker.sol contract (e.g. TokenLockerOnEth.Sol AND TokenLockerOnHarmony.sol)
+Currently only ERC20 are supported. Moving forward ERC721 and ERC1155 as well as operations on smart contracts will also be supported. For now all client transactions will be done using the CLI. Moving forward  the current bridge (bridge.harmony.one) will be migrated to https://bridge-validator-1.web.app/ and jenya also built a fresh frontend for upcoming trustless bridge: https://github.com/harmony-one/horizon-trustless-frontend
+but most likely, we will stick to the first one.
+* Harmony Node: we use a [harmony local node](https://github.com/harmony-one/harmony#dev-docker-image) this can be run via docker. **There is a pull request which needs to be pushed to Harmony Testnet to enable the bridging functionality. [we need this PR to be pushed to testnet (should be done by May 27th, 2022)](https://github.com/harmony-one/harmony/pull/3872)**
+* For fully trustless bridge we need the bls signature verification precompile to be available on ethereum [eip-2537](https://eips.ethereum.org/EIPS/eip-2537), however this won't be a blocker, as we can initially do permissioned relayers, later adopt optimistic approach, etc. there are many fallback plans for this.
+
+**Current Status**
+* End to End Testing : Has never been succesfully completed
+* Smart Contract Tests: Have errors in them.
+
+**Testing Summary**
+
+| #   | Status | Step                          | Notes |
+| --- | ------ | ----------------------------- | ----- |
+| 1   | *PASS  | Infrastructure Setup          | Working with Local Net until [PR 3872](https://github.com/harmony-one/harmony/pull/3872) is pushed to Testnet |
+| 2   | TBD    | Ropsten Smart Contract Deploy | | 
+| 3   | TBD    | Harmony Smart Contract Deploy | |
+| 4   | TBD    | Relayer Running               | |
+| 5   | TBD    | End To End Testing            | |
+
+**Migration Strategy**
+* Smart Contract use Hardhat with Typescript and ethers(instead of web3)
+  * Replace all web3 with ethers
+  * replace all js files with typescript
```

Comment:
Similarly for this - let's make sure it doesn't create too much extra work since we need to launch this fast. Can this be postponed for later?